### ISSUE TITLE
No way to configure "--source-type-override and -source-location-override" with sourceControlType: 'jenkins'

### ISSUE THREAD

**Issue by lkolchin** [awslabs/aws-codebuild-jenkins-plugin#101]
Hello,

When using ```sourceControlType: 'project',``` option all works perfectly.

But when switching to ```sourceControlType: 'jenkins'``` there are a few gotchas.

First of all when creating a project with:
--source type=NO_SOURCE (which works fine with ```sourceControlType: 'project'```)  and then later on trying to work with ```sourceControlType: 'jenkins'```
CodeBuild will scream that - [AWS CodeBuild Plugin] Please select a project with S3 source type
And ```sourceTypeOverride: 'S3'``` doesn't work/exists.

OK, so I've updated the project to have something like this to make it work:
(Yeah and I'm assuming that the requirement for the 'versioned' source bucket - ([AWS CodeBuild Plugin] A versioned S3 bucket is required.) is something that's coming from CodeBuild itself or is it a plugin thing?)
```
--source "{\"type\": \"S3\",\"location\": \"bucketname/prefix/jenkins-sources/source.zip\"}"
```

awsCodeBuild plugin won't allow you to override the following values in its config - "awsCodeBuild ..."  (But I see in the AWS doco that you should be able to with "--source-type-override and -source-location-override"):

So essentially this doesn't work - 
```
sourceControlType: 'jenkins',
sourceTypeOverride: 'S3',
sourceLocationOverride: 'bucketname/prefix/source.zip',
```

If that worked I could have parameterized the file name and it wouldn't have to be source.zip all the time.

Could you guys look into this please?

**Comment by leoherran-aws** [comment]
Thanks for your feedback, we'll work on supporting `sourceTypeOverride` and `sourceLocationOverride` for the `sourceControlType: 'jenkins'` case. We'll keep this issue up to date.

**Comment by lkolchin** [comment]
Appreciate your prompt reply.
I think I'm gonna write a blogpost on your awesome plugin as I think it's a bit underreported :)

**Comment by leoherran-aws** [comment]
Added support for this in 98ef22e23d7425472734664c84259cf07e7c4c1d. This will be released in version `0.51` of the plugin, which should be available in a couple of hours. 

That's awesome, thank you! Be sure to post the link here when it's ready!

**Comment by lkolchin** [comment]
Thank you.
Works a treat!
